source/solvers/heat_transfer.cc

```diff
@@ -977,6 +963,23 @@ HeatTransfer<dim>::postprocess(bool first_iteration)
         }
     }
 
+  if (simulation_parameters.post_processing.calculate_average_temp_and_hf)
+    {
+      // Start calculating after the initial time for the average temperature
+      // and average heat flux
+      if (this->simulation_control->get_current_time() >
+          simulation_parameters.post_processing
+              .initial_time_for_average_temp_and_hf -
+            1e-6 * simulation_control->get_time_step())
+        {
+          this->average_temperature->calculate_average_scalar(
+            this->present_solution,
+            this->simulation_parameters.post_processing,
+            simulation_control->get_current_time(),
+            simulation_control->get_time_step());
+        }
```

Comment:
Shouldnt this logic be inside of the averaging function instead? I'm a bit curious, why si the logic inside of the heat transfer physics when the averaging function also takes into account the time? what I am trying to say is you  could be calling it every time step the averaging and let teh averaging class take care of deciding if it should average or not, like it is done for the velocity averaging. You need to pass the tiem anyway, so why  is the check for time done outside of the function itself?